File: Pythonleetcode/Leetcode144.py
class ListNode(object):
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution(object):
    def helper(self,root,res):
        if not root:
            return
        res.append(root.val)
        self.helper(root.left,res)
        self.helper(root.right, res)


    def preorderTraversal(self, root):
        res=[]
        self.helper(root,res)
        return res

File: CppPractice/CppLeetcode/Leetcode144.cpp
#include<vector>
using namespace std;

struct TreeNode
{
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode():val(0),left(nullptr),right(nullptr){};
    TreeNode(int x):val(x),left(nullptr),right(nullptr){};
    TreeNode(int x, TreeNode *leftp,TreeNode *rightp):val(x),left(leftp),right(rightp){};
};
class Solution {
private:
    void Traversal(TreeNode *p,vector<int> &res)
    {
        if (!p)
        {
            return;
        }
        else
        {
            res.push_back(p->val);
            Traversal(p->left,res);
            Traversal(p->right,res);
        }
    }
public:
    vector<int> preorderTraversal(TreeNode* root)  //Mid/left/right
    {
        vector<int> result;
        Traversal(root,result);
        return result;
    }
};